bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Pattern.java

```diff
@@ -221,27 +235,50 @@ public Pattern(Device device, float x1, float y1, float x2, float y2, Color colo
 			Gdip.LinearGradientBrush_SetInterpolationColors(handle, new int [] {foreColor, midColor, backColor}, new float[]{0, 0.5f, 1}, 3);
 		}
 	}
+	this.handleMap.put(zoomLevel, handle);
 	init();
 }
 
+void setImageHandle(Image image, int zoomLevel) {
+	long[] gdipImage = image.createGdipImage(zoomLevel);
+	long img = gdipImage[0];
+	int width = Gdip.Image_GetWidth(img);
+	int height = Gdip.Image_GetHeight(img);
+	long handle = Gdip.TextureBrush_new(img, Gdip.WrapModeTile, 0, 0, width, height);
+	handleMap.put(zoomLevel, handle);
```

Comment:
Doesn't this mean that you could end up having an entry with `handle == 0` in the map? This could affect the logic since you rely on the map being empty in `isDisposed()` but you allow disposed handles into the map.

You should only add entries to the map `if (handle != 0)` (_i.e._ add an `else` clause the the `if` below)

Reply:
Done